packages/atlas/src/components/_crt/BuyMarketTokenModal/steps/BuyMarketTokenForm.tsx

```diff
@@ -13,62 +14,68 @@ import { DetailsContent } from '@/components/_nft/NftTile'
 import { atlasConfig } from '@/config'
 import { useMediaMatch } from '@/hooks/useMediaMatch'
 import { useMountEffect } from '@/hooks/useMountEffect'
-import { hapiBnToTokenNumber } from '@/joystream-lib/utils'
-import { useJoystream, useSubscribeAccountBalance } from '@/providers/joystream'
+import { hapiBnToTokenNumber, tokenNumberToHapiBn } from '@/joystream-lib/utils'
+import { useFee, useJoystream, useSubscribeAccountBalance } from '@/providers/joystream'
 
 import { CommonProps } from './types'
 
-export const getTokenDetails = (_: string) => ({
-  title: 'JBC',
-  pricePerUnit: 1000,
-  tokensOnSale: 67773,
-  userBalance: 100000,
-})
-
-type BuySaleTokenFormProps = {
-  tokenId: string
+type BuyMarketTokenFormProps = {
+  token?: FullCreatorTokenFragment | null
   onSubmit: (tokens: number | null) => void
+  pricePerUnit: number
 } & CommonProps
 
-export const BuySaleTokenForm = ({ tokenId, setPrimaryButtonProps, onSubmit }: BuySaleTokenFormProps) => {
-  const { control, watch, handleSubmit } = useForm<{ tokens: number | null }>()
+export const BuyMarketTokenForm = ({
+  token,
+  setPrimaryButtonProps,
+  onSubmit,
+  pricePerUnit,
+}: BuyMarketTokenFormProps) => {
+  const { control, watch, handleSubmit, formState } = useForm<{ tokens: number | null }>()
   const { accountBalance } = useSubscribeAccountBalance()
-  const tokens = watch('tokens')
-  const { pricePerUnit, tokensOnSale, userBalance, title } = getTokenDetails(tokenId)
+  const tokens = watch('tokens') || 0
+  const { fullFee } = useFee('purchaseTokenOnMarketTx', ['1', '1', String(tokens ?? 0), '1000000'])
```

Comment:
Did you mean to leave these dummy data ?

Reply:
Yeah, it doesnt really impact fees and calculating it here as well feels like a waste